rosidl_generator_cpp/test/test_array_generator.hpp

```diff
@@ -123,19 +123,19 @@ void test_vector_fill(
   int min, int max,
   int minlength, const int maxlength)
 {
-  if (size > 0 && min != max && minlength != maxlength) {
+  if (size > 0 && min < max && minlength < maxlength) {
     size_t step = (max - min) / size;
     size_t step_length = (maxlength - minlength) / size;
-    char * tmpstr = reinterpret_cast<char *>(malloc(maxlength));
-    std::snprintf(tmpstr, minlength, "%*d", minlength, min);
+    char * tmpstr = reinterpret_cast<char *>(malloc(maxlength + 1));
```

Comment:
Unrelated to the patch: this doesn't need a `reinterpret_cast` - a `static_cast` would be sufficient and "better" here.

Reply:
:+1: https://github.com/ros2/rosidl/pull/275/commits/fee32650c7346022c4b83ee26a43ed2457eb5839